packages/@aws-cdk/aws-amplify/lib/app.ts

```diff
@@ -0,0 +1,347 @@
+      const role = new LazyRole(scope, 'lazy-role', {
+        assumedBy: new ServicePrincipal('amplify.amazonaws.com')
+      });
+
+      roleArn = role.roleArn;
+    }
+
+    const resource = new CfnApp(this, 'App', {
+      accessToken: props.accessToken,
+      basicAuthConfig: this.basicAuthResolver,
+      buildSpec: (props.buildSpec) ? props.buildSpec.toBuildSpec() : undefined,
+      customRules: this.customRulesResolver,
+      description: props.description,
+      environmentVariables: this.environmentVariablesResolver,
+      iamServiceRole: roleArn,
+      name: props.appName,
+      oauthToken: (props.oauthToken && props.oauthToken.toString()) ? props.oauthToken.toString() : undefined,
+      repository: props.repository,
+      tags: props.tags
+    });
+
+    this.appArn = resource.attrArn,
+    this.appDefaultDomain = resource.attrDefaultDomain;
+    this.appId = resource.attrAppId;
+    this.appName = resource.attrAppName;
+
+    this.resource = resource;
+  }
+
+  /**
+   * Add Service Role
+   *
+   * @param role
+   */
+  public addServiceRole(role: IRole) {
+    this.resource.iamServiceRole = role.roleArn;
+  }
+
+  /**
+   * Add Basic Auth on App
+   *
+   * @param username
+   * @param password
+   */
+  public addBasicAuth(username: string, password: string) {
+    this.basicAuthResolver.basicAuthConfig({
+      enableBasicAuth: true,
+      password,
+      username
+    });
+  }
+
+  /**
+   * Add Environment Variable to App
+   *
+   * @param name
+   * @param value
+   */
+  public addEnvironmentVariable(name: string, value: string) {
+    this.environmentVariablesResolver.addEnvironmentVariable(name, value);
+  }
+
+  /**
+   * Add custom rule to app
+   *
+   * @param source
+   * @param target
+   * @param status
+   * @param condition
+   */
+  public addCustomRule(source: string, target: string, status?: string, condition?: string) {
+    this.customRulesResolver.addCustomRule(source, target, status, condition);
+  }
+
+  /**
+   * Add domain to app
+   *
+   * @param id
+   * @param props
+   */
+  public addDomain(id: string, props: DomainBaseProps): Domain {
+    const domainProps = {
+      app: this,
+      ...props
+    };
+
+    return new Domain(this, id, domainProps);
+  }
+
+  /**
+   * Add branch to app
+   *
+   * @param id
+   * @param props
+   */
+  public addBranch(id: string, props: BranchBaseProps): Branch {
+    const branchProps = {
+      app: this,
+      ...props
+    };
+
+    return new Branch(this, id, branchProps);
+  }
+}
+
+/**
+ * App Interface
+ */
+export interface IApp extends IResource {
+  /**
+   * App ID
+   *
+   * @attribute
+   */
+  readonly appId: string;
+}
+
+/**
+ * App Properties
+ */
+export interface AppProps {
+  /**
+   * Personal Access token for 3rd party source control system for an Amplify App, used to create webhook
+   * and read-only deploy key. Token is not stored.
+   *
+   * @default - No access token
+   */
+  readonly accessToken?: string;
```

Comment:
This should be `core.SecretValue` I think. We don't want to encourage people to check-in access tokens.